sys/dev_uart.c

```diff
@@ -22,12 +22,28 @@ static int dev_uart_write(vnode_t *t, uio_t *uio) {
 }
 
 static int dev_uart_read(vnode_t *t, uio_t *uio) {
+  char buffer[UART_BUF_MAX];
+  unsigned curr = 0;
+  while (curr < UART_BUF_MAX && curr < uio->uio_resid) {
+    buffer[curr] = uart_getchar();
```

Comment:
This should not busy-loop but rather use `sleepq_wait`, while interrupt handler should call `sleepq_signal`.